File: src/main/java/org/example/momento/History.java
package org.example.momento;

import java.util.ArrayList;
import java.util.List;

public class History {
    private List<EditorState> states = new ArrayList<>();

    public void push(EditorState editorState) {
        states.add(editorState);
    }

    public EditorState pop() {
        int lastIndex = states.size() - 1;

        EditorState editorState = states.get(lastIndex);

        states.remove(editorState);
        return editorState;
    }
}

File: Kotlin/src/momento/History.kt
package momento

class History {
    private val states:MutableList<EditorState> = mutableListOf()

    fun push(state: EditorState) {
        states.add(state)
    }

    fun pop():EditorState {
        val lastIndex = states.size - 1
        val state = states.get(lastIndex)

        states.removeAt(lastIndex)
        return state

    }
}